### DISCUSSION TITLE
how do I use .keymap file instead of .json?

### DISCUSSION THREAD

**Discussion by szsong** [nickcoutsos/keymap-editor#17] [Q&A] [answered]
The example corne repo worked pretty well....
I have a private repo "zmk-config" and have authorized access to the app. but when switching to that repo, the app reports:
"Missing file config/keymap.json"

I don't have keymap.json file in the repo, only my corne.keymap file... 
Did I miss anything? should I generate a keymap.json file before using the app? Thank you!

**Comment by nickcoutsos** [comment, marked as answer]
Hi!

Sorry I missed this -- I guess I don't get notifications for the discussion feature.

For now, you will need the `keymap.json` file. Reading and writing an existing `.keymap` directly would be very useful but there's a lot of complexity to doing it right. In the meantime ideally the app would support handling a missing/empty keymap but I haven't gotten around to that yet. If you need any help generating the keymap let me know!

edit: hi @szsong, I've made some recent changes to remove the initial barrier of handcrafting a `keymap.json` file. If your repo doesn't include one already the API will generate one with a single empty layer, and if the app is processing a layer with fewer keys than the layout allows it will fill things in with `&none`.
